### PR TITLE
add routines for extracting regions

### PR DESCRIPTION
**Pull request by awalter-bnl**
This adds the functionality required to extract regions from a set of data.

### CODE REVIEW

File: sixtools/preliminary_processing.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        dictionary has the form {'region1_name': region1, ......}
+
+    Examples
+    --------
+    This is intended to be used in the following, one line, call with XXXX
+    being a scan_id or equivalent (start.rixscam.regions is suppose to point
+    to the metadata containg the regions for this scan):
+    >>> extract_regions(db[XXXX].data, db[XXXX].start.rixscam.regions)
+    '''
+    yield pre_proc.extract_regions(arrays, regions)
```
Comment (danielballan) [inline sixtools/preliminary_processing.py]: Should this be `yield from`?

File: sixtools/tests/test_preliminary_processing.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+
+@pytest.fixture
+def data():
+    # generate a list of 2D numpy arrays.
+    out_list = []
+    for i in range(0, 3):
+        out_list.append(np.arange(1, 26).reshape(5, 5))
+
+    yield out_list
```
Comment (danielballan) [inline sixtools/tests/test_preliminary_processing.py]: This construction is odd. I think this should either be:

```py
    for i in range(0, 3):
        yield np.arange(1, 26).reshape(5, 5)
```

or

```py
    out_list = []
    for i in range(0, 3):
        out_list.append(np.arange(1, 26).reshape(5, 5))

    return out_list
```

File: sixtools/preliminary_processing.py (hunk 1 above)
Comment (awalter-bnl) [reply sixtools/preliminary_processing.py]: I think you are correct here, thanks for pointing this out

File: sixtools/tests/test_preliminary_processing.py (hunk 2 above)
Comment (awalter-bnl) [reply sixtools/tests/test_preliminary_processing.py]: it should yield a list of 3 identical arrays at each point. not a single array (as in your first 
ie.
`next (instance)`
`[array1, array2, array3]`
suggestion). It needs to be a generator, so your secodn sugestion is also not what is needed

Comment (awalter-bnl) [reply sixtools/tests/test_preliminary_processing.py]: I should add for clarification, this imitates the output structure for db[xxxx].data(....)

**Comment by awalter-bnl** [comment]
closing this. Will reopen a new one with the merged rixs and sixtools functions

**Pull request closed without merging**